index.js

```diff
@@ -0,0 +1,68 @@
+const readline = require('readline-sync');
+const parse = require('csv-parse/lib/sync');
+const log4js = require('log4js');
+const logger = log4js.getLogger('logs\\debug.log');
+const fs = require('fs');
+const splitPoundsPennies = require('./splitPoundsPennies');
+const createAccountArray = require('./createAccountArray');
+const addSubtractInAccounts = require('./addSubtractInAccounts');
+const parseIntTransacAmounts = require('./parseIntTransacAmounts');
+const recombinePenniesPounds = require('./recombinePenniesPounds');
+const userRequestList = require('./userRequestList');
+
+log4js.configure({
+    appenders: {
+        file: { type: 'fileSync', filename: 'logs\\debug.log' }
+    },
+    categories: {
+        default: { appenders: ['file'], level: 'debug'}
+    }
+});
+
+class Transaction {
+    constructor(Date,FromAccount,ToAccount,Narrative,Amount,amountPounds,amountPennies){
+    this.Date=Date;
+    this.FromAccount=FromAccount;
+    this.ToAccount=ToAccount;
+    this.Narrative=Narrative;
+    this.Amount=Amount;
+    this.amountPounds=amountPounds;
+    this.amountPennies=amountPennies;
+    }
+}
+function runMyProgram() {
+    console.log('\nPlease enter name of file to be used:');
+    const fileChoice=readline.prompt();
+    fileType=fileChoice.split(".")[1];
+
+    if (fileType==='csv') {
+        const fileContents=fs.readFileSync(fileChoice,'utf8');
+        const parsedFile=parse(fileContents);
+                
+        var allTransactions=[];

```

Comment:
Rather than declaring `var allTransactions = ...` inside each branch, it'd be better to declare `let allTransactions;` outside of the if (or switch) and just set it (`allTransactions = ...`) inside the branches. As written, I might skip the contents of the branches when reading the code and be confused as to where `allTransactions` has come from once we start using it.

(The general rule here is that you shouldn't declare variables somewhere deeper than where they're used; it happens to work with `var`, but won't with `let`, and avoiding misleading code like that is part of why `let` is preferable!)